src/lib/dataObjectUtils.ts

```diff
@@ -0,0 +1,441 @@
+      parentPath = parentPath.includes('.')
+        ? parentPath.substring(0, parentPath.lastIndexOf('.'))
+        : '';
+    }
+  }
+  return prune;
+};
+type TargetFunction = (input: string | object) => Record<string, any>;
+type TargetFunctions = { [key:string]: TargetFunction };
+
+/**
+ * Remaps the data based on the provided remapping definition.
+ * @param data - The JSON data to be remapped.
+ * @param remappingDefinition - The remapping definition object.
+ * @param reverse - Flag indicating whether to perform reverse remapping. Default is false.
+ * @returns The remapped data.
+ */
+const remap = (
+  data: object | object[],
+  remappingDefinition: RemappingDefinition,
+  options:{
+    reverse?: boolean,
+    keepUnmappedValues?: boolean,
+    deleteMappedValues?: boolean,
+    deleteEmptyParents?: boolean,
+    targetFunctions?: TargetFunctions,
+  } = {},
+): {
+  mapped: object | object[] | null,
+  unmapped: object | object[] | null,
+} => {
+  // If data is null or undefined, return null
+  if (data === null || data === undefined) return { mapped: null, unmapped: null };
+  const {
+    reverse = false,
+    deleteMappedValues = true,
+    deleteEmptyParents = true,
+    keepUnmappedValues = true,
+    targetFunctions = {},
+  } = options;
+
+  let remappedData;
+  let unmappedData = data;
+  if (keepUnmappedValues) {
+    remappedData = data;
+  } else {
+    const targets = reverse
+      ? Object.keys(remappingDefinition)
+      : Object.values(remappingDefinition);
+    const targetStructureIsArray = targets
+      .every((target) => {
+        if (Array.isArray(target)) return false;
+        const prefix = target.split('.')[0];
+        return (prefix === '*'
+        || (Number.isInteger(Number(prefix))
+        && Number(prefix) > 0));
+      });
+    if (targetStructureIsArray) {
+      remappedData = [];
+    } else {
+      remappedData = {};
+    }
+  }
+
+  // Iterate over each key in the remapping definition
+  Object.keys(remappingDefinition).forEach((key) => {
+    // Determine the source and target paths based on the reverse flag
+    // eslint-disable-next-line no-nested-ternary
+    let sourcePath:string;
+    if (reverse) {
+      sourcePath = Array.isArray(remappingDefinition[key])
+        ? remappingDefinition[key].slice(-1) as string
+        : remappingDefinition[key] as string;
+    } else {
+      sourcePath = key;
+    }
+    const targetDefinition = reverse
+      ? key
+      : remappingDefinition[key];
+    const targetActions = Array.isArray(targetDefinition)
+      ? targetDefinition
+      : [targetDefinition];
+    // Get the value from the source path in the remapped data
+    const sourceValue = dotWild.get(
+      keepUnmappedValues
+        ? remappedData
+        : data,
+      sourcePath,
+    );
+
+    // If the source value exists
+    if (sourceValue !== undefined || sourceValue !== null) {
+      targetActions.forEach((targetAction) => {
+        /* TODO: fix later
```

Comment:
I disagree TODO's are useful even if they don't all get addressed as part of this PR. As they indicate that the code in this region might need to at least have another look over.

Reply:
Marked it as a nit. Mainly because I was institutionalized to reject any PRs that had commented out that code or to do lists. As it was all considered new work and needed to be identified as a potential work item in the backlog. So that's always the first pass two things I look for.